visitors/views.py

```diff
@@ -15,7 +17,42 @@
 from .signals import self_service_visitor_created
 
 
-class SelfService(View):
+class SelfServiceBase(View):
+    """Base view for self-service pages."""
+
+    # default visitor request templates
+    template_name = ""
+
+    def __init__(self, **kwargs: Any) -> None:
+        self.visitor: Visitor | None = None
+        super().__init__(**kwargs)
+
+    def get_template_name(self) -> str:
+        """Return the path to the template to render."""
+        if self.template_name:
+            return self.template_name
+        raise NotImplementedError("Missing template_name property.")
```

Comment:
Override this to apply custom rules to determine which template to show - method has access to `self.request` and `self.visitor`.